aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -1355,6 +1356,22 @@ module aptos_framework::stake {
         add_stake(&validator_2, 1);
     }
 
+    #[test(aptos_framework = @aptos_framework, validator = @0x123)]
+    #[expected_failure(abort_code = 0x10013)]
+    public entry fun test_active_validator_cannot_add_more_stake_than_limit_in_multiple_epochs(
+        aptos_framework: signer,
+        validator: signer,
+    ) acquires OwnerCapability, StakePool, AptosCoinCapabilities, ValidatorConfig, ValidatorPerformance, ValidatorSet {
+        timestamp::set_time_has_started_for_testing(&aptos_framework);
+        initialize(&aptos_framework);
+        // Only 50% voting power increase is allowed in each epoch.
+        staking_config::initialize_for_test(&aptos_framework, 50, 10000, LOCKUP_CYCLE_SECONDS, true, 1, 10, 50);
+        join_test_staking(&aptos_framework, &validator, true);
+        end_epoch();
+        // Add more than 50% limit. This should fail.
+        add_stake(&validator, 77);
```

Comment:
This only tests one round of voting power increase. Can we test multiple rounds? This would expose the bug fixed by this PR.

Reply:
This is actually "two rounds" which reveal the bug because `join_test_staking` is called with `change_epoch=true`. Anyway, I added another epoch to make it clearer.

Reply:
Can you add assertions to ensure the voting power is correct after each end_epoch call?

Reply:
The assertions are added.